python/ray/_private/worker.py

```diff
@@ -687,75 +687,6 @@ def get_objects(self, object_refs: list, timeout: Optional[float] = None):
             debugger_breakpoint,
         )
 
-    @Deprecated
-    def run_function_on_all_workers(self, function: callable):
```

Comment:
why don't we raise a deprecation exception for now? Not sure about the deprecation policy, but it could feel like a sudden deprecation because we don't print warning on `@Deprecated`

Reply:
+1 to print exception (and maybe point to runtime env doc)

Reply:
yeah, we have already done it in 2.3 :) this PR is for 2.4